scripts/sync-oauth-credentials.js

```diff
@@ -2,33 +2,36 @@ const fs = require('fs');
 const { Core } = require('@adobe/aio-sdk');
 const { replaceEnvVar, resolveEnvPath } = require('../lib/env');
 const dotenv = require('dotenv');
+const { context } = require('@adobe/aio-lib-ims');
 
 const keyMap = {
-  client__id: 'OAUTH_CLIENT_ID',
-  client__secrets: 'OAUTH_CLIENT_SECRETS',
-  technical__account__email: 'OAUTH_TECHNICAL_ACCOUNT_EMAIL',
-  technical__account__id: 'OAUTH_TECHNICAL_ACCOUNT_ID',
+  client_id: 'OAUTH_CLIENT_ID',
+  client_secrets: 'OAUTH_CLIENT_SECRETS',
+  technical_account_email: 'OAUTH_TECHNICAL_ACCOUNT_EMAIL',
+  technical_account_id: 'OAUTH_TECHNICAL_ACCOUNT_ID',
   scopes: 'OAUTH_SCOPES',
-  ims__org__id: 'OAUTH_IMS_ORG_ID',
+  ims_org_id: 'OAUTH_IMS_ORG_ID',
 };
 
 const logger = Core.Logger('scripts/sync-oauth-credentials', { level: process.env.LOG_LEVEL || 'info' });
 
 /**
- * Syncs OAUTH environment variables from AIO_ims_contexts_* variables in the .env file.
+ * Syncs OAUTH environment variables from the configured IMS context in the .env file.
  */
-function main() {
+async function main() {
   try {
-    logger.debug('Sync OAUTH env vars from AIO_ims_contexts_* vars in .env file');
+    logger.debug('Sync OAUTH env vars from configured IMS context in .env file');
     const envPath = resolveEnvPath();
     const envVars = dotenv.parse(fs.readFileSync(envPath, 'utf8'));
-    const imsContextEnvVars = parseImsContextEnvVars(envVars);
-    if (imsContextEnvVars.length === 0) {
-      logger.warn('No AIO_ims_contexts_* environment variables found in .env file');
+    const imsContext = await resolveImsS2SContext();
+    if (!imsContext) {
+      logger.warn('IMS server 2 server OAuth credentials are not configured. Did you run `aio app use`?');
```

Comment:
👏 (not sure `aio app add` needs to be also guided for users who hasn't added i/o management)